File: fp-scala/src/main/scala/com/cscie88a/week3/AdoptionService.scala
package com.cscie88a.week3

// TODO complete the method definitions below
object AdoptionService {
  def tryToAdoptNoArg(
      isAdoptionFeePaid: Boolean
    )(
      readyForAdoptionNoArg: () => Boolean
    ): Boolean = ???

  def tryToAdoptOneArg(
      healthCheckDone: Boolean,
      isAdoptionFeePaid: Boolean
    )(
      readyForAdoptionOneArg: Boolean => Boolean
    ): Boolean = ???

}

File: fp-java/src/main/java/cscie88a/week3/AdoptionService.java
package cscie88a.week3;

public class AdoptionService {

    public boolean tryToAdoptNoArg(
            IAdoptable someoneToAdopt, boolean adoptionFeePaid) {
        if (someoneToAdopt.readyForAdoptionNoArg() && adoptionFeePaid) {
            return true;
        } else
            return false;
    }

    public boolean tryToAdoptOneArg(IAdoptable2 someoneToAdopt, boolean healthCheckDone, boolean adoptionFeePaid) {
        if (someoneToAdopt.readyForAdoptionOneArg(healthCheckDone) && adoptionFeePaid) {
            return true;
        } else
            return false;
    }

}
